examples/events.rs

```diff
@@ -29,8 +30,9 @@ async fn run() {
         .expect("deployment failed");
     let mut transfers = instance
         .instance
-        .event("Transfer")
-        .expect("transfer event not found");
+        .event::<_, (Address, Address, U256)>("Transfer")
```

Comment:
I imagined that we would parse the event type from the contract like `event Deposit(address indexed user, address indexed token, uint256 amount, uint32 batchId);` and then I would get this as a rust type. Then I could say *give me all events* and get an `enum { Deposit(...), Withdraw(...), ... }`.

Is that something you had in mind for the future?

Reply:
That would require code generation. This is just the runtime implementation for decoding topics + logs for a given event.

Also, that would be the equivalent of the `all_events` and not `event("Name")`.

Reply:
To answer your question though, yes that is planned.